tests/test_validation_tool.py

```diff
@@ -326,26 +327,78 @@ def test_validation_negative() -> None:
 
 def test_validation_negative_full_output(tmp_path: Path) -> None:
     number = len(
-        list(Path("tests/data/ethernet/frame/invalid").glob("*"))
-        + list(Path("tests/data/ethernet/frame/valid").glob("*"))
+        list(Path("tests/validation_tool/ethernet/frame/invalid").glob("*"))
+        + list(Path("tests/validation_tool/ethernet/frame/valid").glob("*"))
     )
     assert (
         cli(
             [
                 "validate_spec",
                 "-s",
-                "in_ethernet.rflx",
+                "tests/validation_tool/in_ethernet.rflx",
                 "-m",
                 "Ethernet::Frame",
                 "-v",
-                "tests/data/ethernet/frame/invalid",
+                "tests/validation_tool/ethernet/frame/invalid",
                 "-i",
-                "tests/data/ethernet/frame/valid",
+                "tests/validation_tool/ethernet/frame/valid",
                 "-o",
                 f"{tmp_path}/output.json",
                 "--no-verification",
             ]
         )
         == f"{number} messages were classified incorrectly"
     )
-    assert cmp(f"{tmp_path}/output.json", "tests/data/valid_full_output_negative.json")
+    assert cmp(f"{tmp_path}/output.json", "tests/validation_tool/valid_full_output_negative.json")
+
+
+def test_validation_coverage(capsys: Any) -> None:
+    assert (
+        cli(
+            [
+                "validate_spec",
+                "-s",
+                "tests/validation_tool/ethernet.rflx",
+                "-m",
+                "Ethernet::Frame",
+                "-v",
+                "tests/validation_tool/ethernet/frame/valid",
+                "-i",
+                "tests/validation_tool/ethernet/frame/invalid",
+                "--coverage",
+                "--target-coverage=100",
+                "--no-verification",
+            ]
+        )
+        == 0
+    )
+    captured_output = capsys.readouterr()
+    with open("tests/validation_tool/valid_cov_output_ethernet.txt", "r") as f:
+        valid_output = f.read().replace("Directory: .", f"Directory: {os.getcwd()}")
+    assert captured_output.out == valid_output
+
+
+def test_coverage_threshold_missed(capsys: Any) -> None:
```

Comment:
If I use `CaptureFixture[str]` and import `CaptureFixture` from ` _pytest.capture` i get an `unused_import` error from pylint. If I remove the import (obviously) `CaptureFixture` is undefined. Additionaly I have to use `type: ignore` for that import becasue `_pytest.cypture` does not use type hints.
How can I supress mypy and pylint on the same line where the import is? Am I using the right import?

I have to put `CaptureFixture[str]` in quotes, because when used without, the error `CaptureFixture is not subscriptable` is raised.

Reply:
I cannot reproduce the issue. It is probably caused by an old version of pytest (or mypy/pylint). What version do you have? After the following changes all checks are successful for me:

```diff
diff --git a/tests/test_validation_tool.py b/tests/test_validation_tool.py
index 93ec1e1..c93007b 100644
--- a/tests/test_validation_tool.py
+++ b/tests/test_validation_tool.py
@@ -5,7 +5,7 @@ from pathlib import Path
 from typing import Iterator
 
 import pytest
-from _pytest.capture import CaptureFixture  # type: ignore
+from _pytest.capture import CaptureFixture
 from rflx.pyrflx import PyRFLX
 
 from tools.validate_spec import _validate_message, cli
@@ -354,7 +354,7 @@ def test_validation_negative_full_output(tmp_path: Path) -> None:
     assert cmp(f"{tmp_path}/output.json", "tests/validation_tool/valid_full_output_negative.json")
 
 
-def test_validation_coverage(capsys: CaptureFixture) -> None:
+def test_validation_coverage(capsys: CaptureFixture[str]) -> None:
     assert (
         cli(
             [
@@ -380,7 +380,7 @@ def test_validation_coverage(capsys: CaptureFixture) -> None:
     assert captured_output.out == valid_output
 
 
-def test_coverage_threshold_missed(capsys: CaptureFixture) -> None:
+def test_coverage_threshold_missed(capsys: CaptureFixture[str]) -> None:
     assert (
         cli(
             [
```

Reply:
I have `mypy 0.812` and `pylint 2.8.2` and `pytest 5.4.3`. `pytest` is an old version but is constraint to a version < 6 in `setup.py` in RecodFlux (I'm using the same venv for RecodFlux and RecordFlux-specifications). I am also using python 3.9.4

Reply:
IIRC type hints where introduced in `pytest 6`, so this should be the cause. We have the version constraint for pytest because of Componolit/RecordFlux#416. Please check if the issue still occurs for recent versions (of pytest and pyicontract-lint). If not, we should change the dependency constraint in RecordFlux (to `pytest >=6, <7`) to allow using type hints also in RecordFlux.

Reply:
It works now with `CaptureFixture[str]` after upgrading pytest. 5088b56